server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailAdapter.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.mailet.Mail;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+public class MailAdapter {
+    private static final AttributeName ORIGIN_MAILBOXES_ATTRIBUTE_NAME = AttributeName.of("originMailboxes");
+    private static final AttributeName HAS_ATTACHMENT_ATTRIBUTE_NAME = AttributeName.of("hasAttachment");
+    private static final AttributeName OWNER_ATTRIBUTE_NAME = AttributeName.of("owner");
+    private static final AttributeName DELIVERY_DATE_ATTRIBUTE_NAME = AttributeName.of("deliveryDate");
+    private static final AttributeName DELETION_DATE_ATTRIBUTE_NAME = AttributeName.of("deletionDate");
+    private static final AttributeName SUBJECT_ATTRIBUTE_VALUE = AttributeName.of("subject");
+
+    private final MailboxId.Factory mailboxIdFactory;
+    private final MessageId.Factory messageIdFactory;
+
+    public MailAdapter(MailboxId.Factory mailboxIdFactory, MessageId.Factory messageIdFactory) {
+        this.mailboxIdFactory = mailboxIdFactory;
+        this.messageIdFactory = messageIdFactory;
+    }
+
+    Mail toMail(DeletedMessage deletedMessage) throws MessagingException {
+        MimeMessageCopyOnWriteProxy mimeMessage = new MimeMessageCopyOnWriteProxy(new MimeMessageInputStreamSource(deletedMessage.getMessageId().serialize(), deletedMessage.getContent().get()));
+
+        Collection<AttributeValue<?>> serializedMailboxIds = deletedMessage.getOriginMailboxes().stream()
+            .map(MailboxId::serialize)
+            .map(AttributeValue::of)
+            .collect(Guavate.toImmutableList());
+
+        AttributeValue<?> subjectAttributeValue = AttributeValue.of(deletedMessage.getSubject().map(AttributeValue::of));
+
+        return MailImpl.builder()
+            .name(deletedMessage.getMessageId().serialize())
+            .sender(deletedMessage.getSender())
+            .recipients(deletedMessage.getRecipients())
+            .mimeMessage(mimeMessage)
+            .attribute(new Attribute(OWNER_ATTRIBUTE_NAME, AttributeValue.of(new SerializableUser(deletedMessage.getOwner()))))
+            .attribute(new Attribute(HAS_ATTACHMENT_ATTRIBUTE_NAME, AttributeValue.of(deletedMessage.hasAttachment())))
+            .attribute(new Attribute(DELIVERY_DATE_ATTRIBUTE_NAME, AttributeValue.of(new SerializableDate(deletedMessage.getDeliveryDate()))))
+            .attribute(new Attribute(DELETION_DATE_ATTRIBUTE_NAME, AttributeValue.of(new SerializableDate(deletedMessage.getDeletionDate()))))
+            .attribute(new Attribute(ORIGIN_MAILBOXES_ATTRIBUTE_NAME, AttributeValue.of(serializedMailboxIds)))
+            .attribute(new Attribute(SUBJECT_ATTRIBUTE_VALUE, subjectAttributeValue))
+            .build();
+    }
+
+    DeletedMessage fromMail(Mail mail) {
+        MemoizedSupplier<byte[]> byteContent = new MemoizedSupplier<>(Throwing.supplier(
```

Comment:
do you expect to have callers not reading the message content?

Reply:
Yes I do. IE searches on date ranges. etc...